src/Raven.Server/Documents/Queries/Graph/SingleEdgeMatcher.cs

```diff
@@ -86,16 +86,35 @@ private void EdgeMatch(Match match, BlittableJsonReaderObject blittableJsonReade
         private bool EdgeMatchExact(BlittableJsonReaderObject blittableJsonReaderObject, MethodExpression methodExpression)
         {
             if (string.Equals(methodExpression.Name.Value, "exact", StringComparison.OrdinalIgnoreCase) == false)
-                return false;
+                throw new NotSupportedException($"Where clause on edge doesn't support {methodExpression} method expression");
             var arg = methodExpression.Arguments[0] as BinaryExpression;
             var left = arg.Left as FieldExpression;
             var right = arg.Right as ValueExpression;
             if (left == null || right == null)
                 return false;
+            string rightStringValue = null;
+            switch (right.Value)
+            {
+                case ValueTokenType.Parameter:
+                    if (QueryParameters.TryGet(right.Token.Value, out rightStringValue) == false)
+                    {
+                        throw new NotSupportedException($"Where clause with method expression {methodExpression} has parameter {right.Token.Value} which is not of string type.");
+                    }
+                    break;
+                case ValueTokenType.String:
+                    rightStringValue = right.Token.Value;
+                    break;
+                case ValueTokenType.Null:
+                    rightStringValue = null;
+                    break;
+                default:
+                    throw new NotSupportedException($"Where clause with method expression {methodExpression} has parameter {right.Token.Value} which is not of string type.");
+            }
             var value = BlittableJsonTraverser.Default.Read(blittableJsonReaderObject, left.FieldValue);
-            if (value is LazyStringValue lsv && lsv.CompareTo(right.Token.Value) == 0)
+            if (value is LazyStringValue lsv && lsv.CompareTo(rightStringValue) == 0)
```

Comment:
Am I'm missing something? You are still not checking that `null` is valid.
Consider the case where the field itself is null, are you are getting `null` as a arg.
You aren't handling this, but you should.
Also, what about things like `Name != $name` ?
You have handled just the equality case, which isn't right.

Also, `Name = $foo AND Age > 1` is valid.
Also, `exact(Age > 2)` is also valid (in this case, the `exact` is doing nothing, obviously

Reply:
this is not null value this is a null because of the casting is null

Reply:
also exact(Age > 2 ) is strange i wouldn't expect it to work

Reply:
It works for regular queries, should work as well
`value is LazyStringValue lsv` returns false for null values.

So a null field value to a null arg will return false but should return true